src/lib.rs

```diff
@@ -67,3 +67,33 @@ unsafe extern "C" fn clock_gettime(
 
     retval
 }
+
+#[no_mangle]
+unsafe extern "C" fn getrandom(
+    buf: *mut libc::c_void,
+    mut buflen: libc::size_t,
+    flags: libc::c_uint,
+) -> libc::ssize_t {
+    // TODO: is this needed? Maybe just `buflen = buflen.min(libc::ssize_t::MAX)` ?
+    buflen = buflen.min(0x1FFFFFF);
+
+    if flags != 0 {
+        // no flags are supported on 3DS
+        *__errno() = libc::EINVAL;
+        return -1;
+    }
+
+    let ret = ctru_sys::PS_GenerateRandomBytes(buf, buflen as libc::c_uint) as libc::ssize_t;
+    if ret < 0 {
+        // this is kind of a hack, but at least gives some visibility to the
+        // error code returned by PS_GenerateRandomBytes I guess? Another option
```

Comment:
Based on this header file, these error codes aren't compatible:
https://github.com/devkitPro/libctru/blob/master/libctru/include/3ds/result.h

We would probably want to parse the 3ds result and return the appropriate error code.

Reply:
Yeah, `std` implementation seems to just report them as "unexpected error code XYZ" if I set it in errno here. I have some local changes to sort of "parse" those error codes based on https://www.3dbrew.org/wiki/Error_codes but seeing that header, I wonder if we can just leverage ports of those macros instead. Let me see if I can put something in `ctru-sys` for this...

Reply:
Looks nice! :+1: